### PR TITLE
Cleanup of string handling

### PR DESCRIPTION
**Pull request by Tieske**
removing some dependencies and awkward constructions

### CODE REVIEW

**Comment by thibaultCha** [comment]
Sounds good but checkout the tests, some appear to be broken.

On a side note, stringy is definitely on my list of "dependencies to get rid of", it was mainly used for its `strip` method. Since I am not a big fan of having to maintain our own utils libraries (with the testing it requires etc), and because the `refactor/cli` branch introduces Penlight for various utilities, I was thinking of gradually replace `stringy` with Penlight's `stringx`.

**Comment by Tieske** [comment]
I ran into these, while browsing code, but fixed them on Windows, so no local tests. :+1:  for Penlight

**Comment by Tieske** [comment]
locally all tests pass (tested without postgress).

File: kong/core/resolver.lua [hunk 1]
```diff
@@ -1,64 +1,39 @@
...
-    return pattern
+  if request_host:find("*", 1, true) then
+    return "^"..request_host:gsub("%.", "%%."):gsub("*", ".+").."$"
   end
 end
 
 -- Handles pattern-specific characters if any.
 local function create_strip_request_path_pattern(request_path)
-  return string_gsub(request_path, "[%(%)%.%%%+%-%*%?%[%]%^%$]", function(c) return "%"..c end)
+  return request_path:gsub("[%(%)%.%%%+%-%*%?%[%]%^%$]", function(c) return "%"..c end)
```
Comment (thibaultCha) [inline kong/core/resolver.lua]: This involved a lookup on such a hot code path that `gsub(subject, ...)` did not. Same for all of the string functions you removed in this module.

Comment (Tieske) [inline kong/core/resolver.lua]: ran some tests using 100mln iterations, couldn't find a noticeable difference

**Comment by Tieske** [comment]
Replaced by PR #1223 against `refactor/cli`

**Pull request closed without merging**